src/index.js

```diff
@@ -1,2 +1,55 @@
+/* eslint-disable linebreak-style */
 /* eslint-disable react/react-in-jsx-scope, react/jsx-filename-extension */
 /* @jsx createElement */
+
+function createElement(tagName, props, ...children) {
+  const element = document.createElement(tagName);
+
+  // props가 없는 null 이면 {}로서 처리
+  Object.entries(props || {}).forEach(([key, value]) => {
+    // DOM의 속성은 소문자 처리
+    element[key.toLowerCase()] = value;
+  });
+
+  children.flat().forEach((child) => {
+    // 텍스트가 아니라 노드면
+    if (child instanceof Node) {
+      element.appendChild(child);
+    } else element.appendChild(document.createTextNode(child));
+  });
+
+  return element;
+}
+
+const iniCount = 0;
+
+function render(count) {
+  function handleClick(cur) {
+    render(cur + 1);
+  }
+  function onClickHandle(ini) {
+    render(ini);
+  }
+  const element = (
+    <div id="hello" className="greeting">
+      <p>Hello, world!</p>
+      <p>
+        <button type="button" onClick={() => handleClick(count)}>
```

Comment:
```suggestion
        <button type="button" onClick={handleClick}>
```

이렇게 코드를 바꿔주면 어떻게 될까요?

Reply:
**handleClick**의 의미는 onClick을 누를 경우에만 해당함수를 실행시키는 의미이며 인수를 전달하지 못하지만 지금 프로그램에서는 render함수 안에서 이루어지기 떄문에 render함수의 매개변수인 count를 직접 받아주게 만들 수 있습니다.
```
function render(count) {
  function handleClick() {
    render(count + 1);
  }
.
.
}
```

말씀해주신 것으로 여러가지 동작을 하다가 궁금점이 생겼는데요

1.
**handleClick()**으로 바꾸어준다면 onClick을 할 때 함수를 실행시키는 것이 아니라 렌더링을 읽다가 도중에 handleClick 함수를 계속해서 실행시키면서 그 안의 render를 계속 불러오면서 무한반복에 빠지는 것으로 파악했습니다. 그럴 경우에 해결책으로 handleClick으로 바꾸거나 
```
function handleClick() {
    return function() {
       redner(count+1);
   }
}
```
위처럼 삼단함수로 바꾸는 걸로 알고 있는데요. eslint에서는 Unexpected unnamed function [func-names](https://eslint.org/docs/rules/func-names)으로 제한을 걸더라구요. 이런경우에는 es6의 화살표함수만이 유일한 해법인가요?

2.
JSX의 중괄호({ }) 안에는 유효한 모든 JavaScript 표현식을 넣을 수 있습니다.[링크](https://ko.reactjs.org/docs/introducing-jsx.html)
중괄호 안에 어떻게 자바스크립트 표현식만이 사용가능한가? 와 같은 항목은 문법적인 사항으로 그냥 암기하고 넘어가면 되는걸까요?

Reply:
> 1. 첫번째 질문 ···

네 말씀하신대로 함수를 바로 전달해주거나 화살표 함수를 사용하여 즉시실행 함수로 변경해주는 2가지 방법이 있어요.


> 2. 두번째 질문 ···

네 사용법에 관한 것은 사용법만 익히셔서 사용하시면 될 것 같아요 ;)